src/sentry/workflow_engine/handlers/action/notification.py

```diff
@@ -1,17 +1,54 @@
+import logging
+from typing import Protocol
+
+from sentry.grouping.grouptype import ErrorGroupType
+from sentry.issues.grouptype import MetricIssuePOC
+from sentry.utils.registry import NoRegistrationExistsError, Registry
 from sentry.workflow_engine.models import Action, Detector
 from sentry.workflow_engine.registry import action_handler_registry
 from sentry.workflow_engine.types import ActionHandler, WorkflowJob
 
+logger = logging.getLogger(__name__)
+
+
+class NotificationHandler(Protocol):
```

Comment:
I don't see a ton of value defining this as a protocol, since it only has a single method and it's a hard contract for implementing classes to abide by. It may be better to define this as an ABC. It may also be a good idea to have another method with a more specific name be the method that subclasses override (for example `def handle_workflow_action`), and have the base `NotificationHandler` class just implement `__call__` as a shim to this method.

This also lets us hoist additional logic like metric collection into the `__call__` implementation in a way that is universal for implementing classes.